framework/src/media/audio/resample/samplerate.c

```diff
@@ -0,0 +1,550 @@
+/******************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+/*
+** Copyright (c) 2002-2016, Erik de Castro Lopo <erikd@mega-nerd.com>
+** All rights reserved.
+**
+** This code is released under 2-clause BSD license. Please see the
+** file at : https://github.com/erikd/libsamplerate/blob/master/COPYING
+*/
+
+#include	<stdio.h>
+#include	<stdint.h>
+#include	<stdlib.h>
+#include	<string.h>
+#include	"samplerate.h"
+#include	"common.h"
+
+
+/**
+ * @structure resampler: main structure used for SRC, it contains context
+ *            variables used between src_simple() calls.
+ * @brief It's internal structure, user can only get the handler via src_init().
+ */
+struct resampler {
```

Comment:
basic style is below(_s as a postfix for structure)
struct resampler_s {
};
typedef struct resampler_s resampler_t